mule-deployer/src/main/java/org/mule/tools/model/ArtifactDescription.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Mule ESB Maven Tools
+ * <p>
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * <p>
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.tools.model;
+
+import org.apache.commons.lang.StringUtils;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+
+
+public class ArtifactDescription {
+
+    private final String COORDINATES_SEPARATOR = ":";
+    private String groupId;
+    private String artifactId;
+    private String version;
+    private String type;
+
+    public ArtifactDescription() {
+
+    }
+
+    public ArtifactDescription(String artifactCoordinates) {
+        checkArgument(StringUtils.isNotEmpty(artifactCoordinates), "The artifact coordinates must not be null nor empty");
+        String[] coordinateElements = artifactCoordinates.split(COORDINATES_SEPARATOR);
+        checkState(coordinateElements.length == 4, "The artifact coordinates are not in the format groupId:artifactId:version:type");
+        setGroupId(coordinateElements[0]);
```

Comment:
Actually, I don't think so because it is not possible to invoke **this** after any other statement in the constructor